zerver/lib/test_classes.py

```diff
@@ -478,7 +478,15 @@ def client_delete(
         django_client = self.client  # see WRAPPER_COMMENT
         self.set_http_headers(extra, skip_user_agent)
         result = django_client.delete(
-            url, encoded, follow=follow, secure=secure, headers=headers, **extra
+            url,
+            encoded,
+            follow=follow,
+            secure=secure,
+            headers={
+                "Content-Type": "application/x-www-form-urlencoded",  # https://code.djangoproject.com/ticket/33230
```

Comment:
I read https://github.com/django/django/pull/17404 and https://code.djangoproject.com/ticket/33230, and it definitely seems upstream made the wrong determination, as noted by the last comment by Lily Foote on the Django issue thread. This workaround isn't expensive to maintain, but might be worth posting on their forum about it for more visiblity.